src/multiprogress.rs

```diff
@@ -44,15 +44,23 @@ impl MultiProgress {
     ///
     /// The progress bar will be positioned below all other bars in the [`MultiProgress`].
     pub fn add(&self, pb: ProgressBar) -> ProgressBar {
-        let bars_count = self.bars.read().unwrap().len();
+        let bars_count = self.length();
         self.insert(bars_count, pb)
     }
 
+    /// Inserts a progress bar before the previous one and returns an internalized reference to it.
+    ///
+    /// The progress bar will be positioned before the previous progress bar in the [`MultiProgress`].
+    pub fn insert_before_the_last(&self, pb: ProgressBar) -> ProgressBar {
```

Comment:
I mean, but we already have an `insert()`, you don't need to create such a highly specialized method, instead you just have to expose/create a `len()` method to figure out where to insert. What if tomorrow somebody come here with insert_after_the_first method? And so on